File: scripts/dock2tauri.js
#!/usr/bin/env node

/**
 * Dock2Tauri Node.js Launcher
 * Transform any Docker container into a native desktop application using Tauri.
 * 
 * Usage: node dock2tauri.js <docker-image> <host-port> <container-port>
 */

const fs = require('fs');
const path = require('path');
const os = require('os');
const { spawn, exec } = require('child_process');
const http = require('http');
const { promisify } = require('util');

// Colors for console output
const colors = {
    red: '\x1b[31m',
    green: '\x1b[32m',
    yellow: '\x1b[33m',
    blue: '\x1b[34m',
    reset: '\x1b[0m'
};

// Logger utility
const logger = {
    info: (msg) => console.log(`${colors.blue}ℹ️  ${msg}${colors.reset}`),
    success: (msg) => console.log(`${colors.green}✅ ${msg}${colors.reset}`),
    error: (msg) => console.log(`${colors.red}❌ ${msg}${colors.reset}`),
    warning: (msg) => console.log(`${colors.yellow}⚠️  ${msg}${colors.reset}`)
};

class Dock2Tauri {
    constructor(image, hostPort, containerPort, buildRelease = false, buildTarget = null) {
        this.image = image || 'nginx:alpine';
        this.hostPort = String(hostPort || 8088);
        this.containerPort = String(containerPort || 80);
        this.containerName = `dock2tauri-${this.image.replace(/[^a-zA-Z0-9]/g, '-')}-${this.hostPort}`;
        this.containerId = null;
        this.scriptDir = __dirname;
        this.buildRelease = buildRelease;
        this.buildTarget = buildTarget;
        this.healthUrl = null;
        this.timeout = 30;
        this.tauriConfigPath = null;
    }

    /**
     * Check if required dependencies are available
     */
    async checkDependencies() {
        logger.info('Checking dependencies...');

        try {
            // Check Docker
            await this.execCommand('docker --version');
            
            // Check if Docker daemon is running
            await this.execCommand('docker info');
            
            // Check Rust/Cargo (optional)
            try {
                await this.execCommand('cargo --version');
            } catch (e) {
                logger.warning('Rust/Cargo not found. Some features may not work.');
            }

            logger.success('Dependencies check passed');
            return true;
        } catch (error) {
            if (error.message.includes('docker')) {
                logger.error('Docker not found or not running. Please install and start Docker.');
            } else {
                logger.error(`Dependency check failed: ${error.message}`);
            }
            return false;
        }
    }

    /**
     * Execute shell command and return result
     */
    execCommand(command) {
        return new Promise((resolve, reject) => {
            exec(command, (error, stdout, stderr) => {
                if (error) {
                    reject(new Error(stderr || error.message));
                } else {
                    resolve(stdout.trim());
                }
            });
        });
    }

    /**
     * Stop existing containers using the same port
     */
    async stopExistingContainers() {
        logger.info(`Stopping existing containers on port ${this.hostPort}...`);

        try {
            // Get containers using the same port
            const result = await this.execCommand(`docker ps -q --filter "publish=${this.hostPort}"`);
            
            if (result) {
                const containerIds = result.split('\n').filter(id => id.trim());
                for (const containerId of containerIds) {
                    await this.execCommand(`docker stop ${containerId}`);
                }
                logger.success('Stopped existing containers');
            }

            // Remove old container with same name
            try {
                const containers = await this.execCommand(`docker ps -a --filter "name=${this.containerName}"`);
                if (containers.includes(this.containerName)) {
                    await this.execCommand(`docker rm ${this.containerName}`);
                    logger.success(`Removed old container: ${this.containerName}`);
                }
            } catch (e) {
                // Container doesn't exist, which is fine
            }

        } catch (error) {
            logger.warning(`Error during cleanup: ${error.message}`);
        }
    }

    /**
     * If `this.image` is a Dockerfile path, build it and set `this.image` to a local tag.
     */
    async prepareImageOrDockerfile() {
        try {
            if (this.image && fs.existsSync(this.image) && fs.statSync(this.image).isFile()) {
                const dockerfilePath = this.image;
                const ctx = path.dirname(dockerfilePath);
                const base = path.basename(dockerfilePath).toLowerCase().replace(/[^a-z0-9_.-]/g, '-');
                const tag = `dock2tauri-local-${base}-${Date.now()}`;
                logger.info(`Building Docker image from ${dockerfilePath} (context: ${ctx}) as ${tag} ...`);
                await this.execCommand(`docker build -f "${dockerfilePath}" -t "${tag}" "${ctx}"`);
                this.image = tag;
                this.containerName = `dock2tauri-${this.image.replace(/[^a-zA-Z0-9]/g, '-')}-${this.hostPort}`;
            }
        } catch (error) {
            logger.error(`Failed to build Dockerfile: ${error.message}`);
            throw error;
        }
    }

    /**
     * Launch the Docker container
     */
    async launchContainer() {
        logger.info('Launching Docker container...');
        logger.info(`Image: ${this.image}`);
        logger.info(`Host Port: ${this.hostPort}`);
        logger.info(`Container Port: ${this.containerPort}`);

        try {
            const command = [
                'docker', 'run', '-d',
                '-p', `${this.hostPort}:${this.containerPort}`,
                '--name', this.containerName,
                '--restart', 'unless-stopped',
                this.image
            ].join(' ');

            this.containerId = await this.execCommand(command);
            logger.success(`Container launched: ${this.containerId}`);
            logger.success(`Access at: http://localhost:${this.hostPort}`);
            return true;

        } catch (error) {
            logger.error('Failed to launch container');
            logger.error(`Error: ${error.message}`);
            return false;
        }
    }

    /**
     * Wait for service to be ready
     */
    async waitForService() {
        logger.info('Waiting for service to be ready...');

        const url = this.healthUrl || `http://localhost:${this.hostPort}`;
        const timeout = Number(this.timeout) || 30;
        for (let i = 0; i < timeout; i++) {
            try {
                await this.checkUrl(url);
                logger.success('Service is ready!');
                return true;
            } catch (e) {
                process.stdout.write('.');
                await this.sleep(1000);
            }
        }

        console.log(); // New line
        logger.warning('Service might not be ready yet, but continuing...');
        return false;
    }

    /**
     * Check if URL is accessible
     */
    checkUrl(url) {
        return new Promise((resolve, reject) => {
            const request = http.get(url, { timeout: 1000 }, (res) => {
                resolve(res.statusCode === 200);
            });
            
            request.on('error', reject);
            request.on('timeout', () => {
                request.destroy();
                reject(new Error('Timeout'));
            });
        });
    }

    /**
     * Sleep for specified milliseconds
     */
    sleep(ms) {
        return new Promise(resolve => setTimeout(resolve, ms));
    }

    /**
     * Generate ephemeral Tauri configuration and store its path in this.tauriConfigPath
     */
    generateTauriConfig() {
        logger.info('Preparing Tauri configuration (ephemeral)...');

        const config = {
            "$schema": "../node_modules/@tauri-apps/cli/schema.json",
            "productName": `Dock2Tauri - ${this.image.split(':')[0].replace(/[\/\\:*?"<>|]/g, '')}`,
            "version": "1.0.0",
            "identifier": `com.dock2tauri.${this.image.replace(/[^a-zA-Z0-9]/g, '')}`,
            "build": {
                "beforeBuildCommand": "",
                "beforeDevCommand": "",
                "devUrl": `http://localhost:${this.hostPort}`,
                "frontendDist": "../app"
            },
            "app": {
                "security": {
                    "csp": null
                },
                "windows": [{
                    "title": `Dock2Tauri - ${this.image}`,
                    "width": 1200,
                    "height": 800,
                    "minWidth": 600,
                    "minHeight": 400,
                    "resizable": true,
                    "fullscreen": false
                }]
            },
            "bundle": {
                "active": true,
                "targets": ["appimage", "deb", "rpm"],
                "icon": [],
                "resources": [],
                "externalBin": [],
                "copyright": "",
                "category": "DeveloperTool",
                "shortDescription": "Docker App in Tauri",
                "longDescription": `Running ${this.image} as desktop application`
            },
            "plugins": {}
        };

        const tmpPath = path.join(os.tmpdir(), `tauri.conf.${Date.now()}.${Math.random().toString(36).slice(2)}.json`);
        try {
            fs.writeFileSync(tmpPath, JSON.stringify(config, null, 2));
            this.tauriConfigPath = tmpPath;
            logger.success(`Ephemeral Tauri configuration prepared at ${tmpPath}`);
            return true;
        } catch (error) {
            logger.error(`Failed to prepare Tauri config: ${error.message}`);
            return false;
        }
    }

    /**
     * Launch Tauri application
     */
    async launchTauri() {
        if (this.buildRelease) {
            logger.info('Building Tauri release bundles (cargo tauri build)...');
        } else {
            logger.info('Launching Tauri application (dev)...');
        }

        const tauriDir = path.join(this.scriptDir, '..', 'src-tauri');
        
        return new Promise((resolve, reject) => {
            // Change to tauri directory
            process.chdir(tauriDir);

            // Try Tauri CLI first, then cargo
            let command, args;
            
            // Check if cargo command exists
            exec('which cargo', (error) => {
                if (error) {
                    logger.error('Cargo not found. Please install Rust and Tauri CLI.');
                    logger.info(`Container is running at: http://localhost:${this.hostPort}`);
                    logger.info(`Container ID: ${this.containerId}`);
                    reject(error);
                    return;
                }

                command = 'cargo';
                if (this.buildRelease) {
                    args = ['tauri', 'build'];
                    if (this.buildTarget) {
                        args.push('--target', this.buildTarget);
                    }
                } else {
                    args = ['tauri', 'dev'];
                }

                if (this.tauriConfigPath) {
                    args.push('--config', this.tauriConfigPath);
                }

                const tauriProcess = spawn(command, args, {
                    stdio: 'inherit',
                    shell: true
                });

                tauriProcess.on('error', (error) => {
                    logger.error(`Failed to launch Tauri: ${error.message}`);
                    logger.info(`Container is running at: http://localhost:${this.hostPort}`);
                    logger.info(`Container ID: ${this.containerId}`);
                    reject(error);
                });

                tauriProcess.on('exit', (code) => {
                    if (code === 0) {
                        logger.info('Tauri application exited normally');
                        resolve(true);
                    } else {
                        logger.warning(`Tauri application exited with code ${code}`);
                        resolve(false);
                    }
                });

                // Handle Ctrl+C gracefully
                process.on('SIGINT', () => {
                    logger.info('Application interrupted by user');
                    tauriProcess.kill('SIGINT');
                    resolve(true);
                });
            });
        });
    }

    /**
     * Cleanup resources
     */
    async cleanup() {
        logger.info('Cleaning up...');

        // Stop and remove container
        if (this.containerId) {
            try {
                await this.execCommand(`docker stop ${this.containerId}`);
                await this.execCommand(`docker rm ${this.containerId}`);
                logger.success('Container stopped and removed');
            } catch (error) {
                logger.warning(`Error during container cleanup: ${error.message}`);
            }
        }

        // Remove ephemeral config
        if (this.tauriConfigPath && fs.existsSync(this.tauriConfigPath)) {
            try {
                fs.unlinkSync(this.tauriConfigPath);
                logger.success('Removed ephemeral Tauri config');
            } catch (e) {
                logger.warning(`Failed to remove ephemeral config: ${e.message}`);
            }
        }
    }

    /**
     * Main execution flow
     */
    async run() {
        console.log(`${colors.blue}🐳🦀 Dock2Tauri - Docker to Desktop Bridge${colors.reset}`);
        console.log('==================================================');

        try {
            if (!(await this.checkDependencies())) {
                return false;
            }
            await this.prepareImageOrDockerfile();
            await this.stopExistingContainers();

            if (!(await this.launchContainer())) {
                return false;
            }

            await this.waitForService();
            this.generateTauriConfig();

            // Launch Tauri (this will block until app exits)
            await this.launchTauri();

        } catch (error) {
            if (error.message === 'SIGINT') {
                logger.info('Interrupted by user');
            } else {
                logger.error(`Unexpected error: ${error.message}`);
                return false;
            }
        } finally {
            await this.cleanup();
        }

        return true;
    }
}

/**
 * Show help information
 */
function showHelp() {
    console.log('Dock2Tauri - Docker to Desktop Bridge');
    console.log('');
    console.log('Usage: node dock2tauri.js [IMAGE|Dockerfile] [HOST_PORT] [CONTAINER_PORT] [OPTIONS]');
    console.log('');
    console.log('Arguments:');
    console.log('  IMAGE|Dockerfile  Docker image to run OR path to Dockerfile (default: nginx:alpine)');
    console.log('  HOST_PORT       Host port to bind to (default: 8088)');
    console.log('  CONTAINER_PORT  Container port to expose (default: 80)');
    console.log('');
    console.log('Options:');
    console.log('  --build (-b)       Build Tauri release bundles instead of running dev mode');
    console.log('  --target=<triple>  Pass target triple to cargo tauri build');
    console.log('  --health-url=<url> Override readiness URL (default: http://localhost:HOST_PORT)');
    console.log('  --timeout=<seconds> Readiness timeout in seconds (default: 30)');
    console.log('');
    console.log('Examples:');
    console.log('  node dock2tauri.js nginx:alpine 8088 80');
    console.log('  node dock2tauri.js grafana/grafana 3001 3000 --build');
    console.log('  node dock2tauri.js jupyter/scipy-notebook 8888 8888 --target=x86_64-pc-windows-gnu');
    console.log('');
    console.log('Environment Variables:');
    console.log('  DOCK2TAURI_DEBUG=1    Enable debug mode');
}

/**
 * Main function
 */
async function main() {
    // Parse command line arguments
    const args = process.argv.slice(2);

    // Handle help
    if (args.includes('-h') || args.includes('--help')) {
        showHelp();
        process.exit(0);
    }

    // Handle Ctrl+C gracefully
    process.on('SIGINT', () => {
        console.log('\nReceived SIGINT, cleaning up...');
        process.exit(0);
    });

    // Parse flags
    let buildRelease = false;
    let buildTarget = null;
    let healthUrl = null;
    let timeout = 30;
    const positionalArgs = [];

    for (const arg of args) {
        if (arg === '--build' || arg === '-b') {
            buildRelease = true;
        } else if (arg.startsWith('--target=')) {
            buildTarget = arg.split('=')[1];
        } else if (arg.startsWith('--health-url=')) {
            healthUrl = arg.split('=')[1];
        } else if (arg.startsWith('--timeout=')) {
            const v = parseInt(arg.split('=')[1], 10);
            if (!Number.isNaN(v)) timeout = v;
        } else if (!arg.startsWith('-')) {
            positionalArgs.push(arg);
        }
    }

    // Extract positional arguments
    const [image, hostPort, containerPort] = positionalArgs;

    if (positionalArgs.length === 0) {
        logger.warning('No arguments provided, using defaults');
    }

    console.log(`${colors.blue}🐳🦀 Dock2Tauri - Docker to Desktop Bridge${colors.reset}`);
    console.log('==================================================');

    // Create and run Dock2Tauri instance
    const dock2tauri = new Dock2Tauri(image, hostPort, containerPort, buildRelease, buildTarget);
    if (healthUrl) dock2tauri.healthUrl = healthUrl;
    if (timeout) dock2tauri.timeout = timeout;
    
    try {
        const success = await dock2tauri.run();
        process.exit(success ? 0 : 1);
    } catch (error) {
        logger.error(`Fatal error: ${error.message}`);
        process.exit(1);
    }
}

// Run main function
if (require.main === module) {
    main().catch((error) => {
        logger.error(`Fatal error: ${error.message}`);
        process.exit(1);
    });
}

module.exports = Dock2Tauri;

File: scripts/dock2tauri.py
#!/usr/bin/env python3

"""
Dock2Tauri Python Launcher
Transform any Docker container into a native desktop application using Tauri.

Usage: python dock2tauri.py --image nginx:alpine --host-port 8088 --container-port 80
"""

import argparse
import json
import os
import subprocess
import sys
import time
import urllib.request
from pathlib import Path

class Colors:
    RED = '\033[0;31m'
    GREEN = '\033[0;32m'
    YELLOW = '\033[1;33m'
    BLUE = '\033[0;34m'
    NC = '\033[0m'  # No Color

class Logger:
    @staticmethod
    def info(message):
        print(f"{Colors.BLUE}ℹ️  {message}{Colors.NC}")

    @staticmethod
    def success(message):
        print(f"{Colors.GREEN}✅ {message}{Colors.NC}")

    @staticmethod
    def error(message):
        print(f"{Colors.RED}❌ {message}{Colors.NC}")

    @staticmethod
    def warning(message):
        print(f"{Colors.YELLOW}⚠️  {message}{Colors.NC}")

class Dock2Tauri:
    def __init__(self, image, host_port, container_port, build_release=False, build_target=None):
        self.image = image
        self.host_port = str(host_port)
        self.container_port = str(container_port)
        self.container_name = f"dock2tauri-{image.replace(':', '-').replace('/', '-')}-{host_port}"
        self.container_id = None
        self.script_dir = Path(__file__).parent
        self.base_dir = self.script_dir.parent
        self.config_file = self.base_dir / "src-tauri" / "tauri.conf.json"
        self.build_release = build_release
        self.build_target = build_target

    def check_dependencies(self):
        """Check if required dependencies are available."""
        Logger.info("Checking dependencies...")
        
        if not self._command_exists("docker"):
            Logger.error("Docker not found. Please install Docker first.")
            return False
        
        if not self._command_exists("cargo"):
            Logger.warning("Rust/Cargo not found. Some features may not work.")
        
        try:
            subprocess.run(["docker", "info"], check=True, capture_output=True)
        except subprocess.CalledProcessError:
            Logger.error("Docker daemon not running. Please start Docker.")
            return False
        
        Logger.success("Dependencies check passed")
        return True

    def _command_exists(self, command):
        """Check if a command exists in the system PATH."""
        return subprocess.run(
            ["which", command], 
            capture_output=True, 
            text=True
        ).returncode == 0

    def stop_existing_containers(self):
        """Stop containers using the same port."""
        Logger.info(f"Stopping existing containers on port {self.host_port}...")
        
        try:
            # Get containers using the same port
            result = subprocess.run([
                "docker", "ps", "-q", "--filter", f"publish={self.host_port}"
            ], capture_output=True, text=True)
            
            if result.stdout.strip():
                container_ids = result.stdout.strip().split('\n')
                for container_id in container_ids:
                    subprocess.run(["docker", "stop", container_id], 
                                 capture_output=True)
                Logger.success("Stopped existing containers")
            
            # Remove old container with same name
            result = subprocess.run([
                "docker", "ps", "-a", "--filter", f"name={self.container_name}"
            ], capture_output=True, text=True)
            
            if self.container_name in result.stdout:
                subprocess.run(["docker", "rm", self.container_name], 
                             capture_output=True)
                Logger.success(f"Removed old container: {self.container_name}")
                
        except Exception as e:
            Logger.warning(f"Error during cleanup: {e}")

    def launch_container(self):
        """Launch the Docker container."""
        Logger.info("Launching Docker container...")
        Logger.info(f"Image: {self.image}")
        Logger.info(f"Host Port: {self.host_port}")
        Logger.info(f"Container Port: {self.container_port}")
        
        try:
            result = subprocess.run([
                "docker", "run", "-d",
                "-p", f"{self.host_port}:{self.container_port}",
                "--name", self.container_name,
                "--restart", "unless-stopped",
                self.image
            ], capture_output=True, text=True, check=True)
            
            self.container_id = result.stdout.strip()
            Logger.success(f"Container launched: {self.container_id}")
            Logger.success(f"Access at: http://localhost:{self.host_port}")
            return True
            
        except subprocess.CalledProcessError as e:
            Logger.error("Failed to launch container")
            Logger.error(f"Error: {e.stderr}")
            return False

    def wait_for_service(self):
        """Wait for the service to be ready."""
        Logger.info("Waiting for service to be ready...")
        
        for i in range(30):
            try:
                with urllib.request.urlopen(
                    f"http://localhost:{self.host_port}", 
                    timeout=1
                ) as response:
                    if response.status == 200:
                        Logger.success("Service is ready!")
                        return True
            except:
                print(".", end="", flush=True)
                time.sleep(1)
        
        print()  # New line
        Logger.warning("Service might not be ready yet, but continuing...")
        return False

    def update_tauri_config(self):
        """Update Tauri configuration with Docker URL."""
        Logger.info("Updating Tauri configuration...")
        
        # Create backup
        backup_file = self.config_file.with_suffix('.json.backup')
        if self.config_file.exists():
            import shutil
            shutil.copy2(self.config_file, backup_file)
        
        config = {
            "$schema": "../node_modules/@tauri-apps/cli/schema.json",
            "productName": f"Dock2Tauri - {''.join(c for c in self.image.split(':')[0] if c not in '/\\:*?\"<>|')}",
            "version": "1.0.0",
            "identifier": f"com.dock2tauri.{''.join(c for c in self.image if c.isalnum())}",
            "build": {
                "beforeBuildCommand": "",
                "beforeDevCommand": "",
                "devUrl": f"http://localhost:{self.host_port}",
                "frontendDist": "../app"
            },
            "app": {
                "security": {
                    "csp": None
                },
                "windows": [{
                    "title": f"Dock2Tauri - {self.image}",
                    "width": 1200,
                    "height": 800,
                    "minWidth": 600,
                    "minHeight": 400,
                    "resizable": True,
                    "fullscreen": False
                }]
            },
            "bundle": {
                "active": True,
                "targets": ["appimage", "deb", "rpm"],
                "icon": [],
                "resources": [],
                "externalBin": [],
                "copyright": "",
                "category": "DeveloperTool",
                "shortDescription": "Docker App in Tauri",
                "longDescription": f"Running {self.image} as desktop application"
            },
            "plugins": {}
        }
        
        try:
            with open(self.config_file, 'w') as f:
                json.dump(config, f, indent=2)
            Logger.success("Tauri configuration updated")
            return True
        except Exception as e:
            Logger.error(f"Failed to update Tauri config: {e}")
            return False

    def launch_tauri(self):
        """Launch the Tauri application."""
        if self.build_release:
            Logger.info("Building Tauri release bundles (cargo tauri build)...")
        else:
            Logger.info("Launching Tauri application (dev)...")
        
        tauri_dir = self.script_dir.parent / "src-tauri"
        
        try:
            os.chdir(tauri_dir)
            
            # Build command based on flags
            if self.build_release:
                if self.build_target:
                    cmd = ["cargo", "tauri", "build", "--target", self.build_target]
                else:
                    cmd = ["cargo", "tauri", "build"]
            else:
                cmd = ["cargo", "tauri", "dev"]
            
            # Try Tauri CLI first
            if self._command_exists("cargo"):
                subprocess.run(cmd, check=True)
            else:
                Logger.error("Cargo not found. Please install Rust and Tauri CLI.")
                Logger.info(f"Container is running at: http://localhost:{self.host_port}")
                Logger.info(f"Container ID: {self.container_id}")
                return False
                
        except subprocess.CalledProcessError as e:
            Logger.error(f"Failed to launch Tauri: {e}")
            return False
        except KeyboardInterrupt:
            Logger.info("Application interrupted by user")
            return True
        
        return True

    def cleanup(self):
        """Clean up resources."""
        Logger.info("Cleaning up...")
        
        if self.container_id:
            try:
                subprocess.run(["docker", "stop", self.container_id], 
                             capture_output=True)
                subprocess.run(["docker", "rm", self.container_id], 
                             capture_output=True)
                Logger.success("Container stopped and removed")
            except Exception as e:
                Logger.warning(f"Error during container cleanup: {e}")
        
        # Restore config backup
        backup_file = self.config_file.with_suffix('.json.backup')
        if backup_file.exists():
            backup_file.rename(self.config_file)
            Logger.success("Tauri configuration restored")

    def run(self):
        """Main execution flow."""
        print(f"{Colors.BLUE}🐳🦀 Dock2Tauri - Docker to Desktop Bridge{Colors.NC}")
        print("==================================================")
        
        try:
            if not self.check_dependencies():
                return False
            
            self.stop_existing_containers()
            
            if not self.launch_container():
                return False
            
            self.wait_for_service()
            self.update_tauri_config()
            
            # Launch Tauri (this will block until app exits)
            self.launch_tauri()
            
        except KeyboardInterrupt:
            Logger.info("Interrupted by user")
        except Exception as e:
            Logger.error(f"Unexpected error: {e}")
            return False
        finally:
            self.cleanup()
        
        return True

def main():
    parser = argparse.ArgumentParser(
        description="Dock2Tauri - Transform Docker containers into desktop apps"
    )
    parser.add_argument(
        "--image", "-i",
        default="nginx:alpine",
        help="Docker image to run (default: nginx:alpine)"
    )
    parser.add_argument(
        "--host-port", "-p",
        type=int,
        default=8088,
        help="Host port to bind to (default: 8088)"
    )
    parser.add_argument(
        "--container-port", "-c",
        type=int,
        default=80,
        help="Container port to expose (default: 80)"
    )
    parser.add_argument(
        "--build",
        action="store_true",
        help="Build Tauri application"
    )
    parser.add_argument(
        "--target",
        help="Build target for Tauri application"
    )
    parser.add_argument(
        "--debug",
        action="store_true",
        help="Enable debug mode"
    )
    
    args = parser.parse_args()
    
    if args.debug:
        import logging
        logging.basicConfig(level=logging.DEBUG)
    
    # Create and run Dock2Tauri instance
    dock2tauri = Dock2Tauri(args.image, args.host_port, args.container_port, args.build, args.target)
    
    success = dock2tauri.run()
    sys.exit(0 if success else 1)

if __name__ == "__main__":
    main()
